### ISSUE TITLE
Dynamic friction model has off-by-one-bug

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#1773]

**Comment by bangerth** [comment]
The dynamic_friction model has a bug (I believe) with this piece of code around line 92 (with my minor change in #1772):
```
      for (unsigned int i = 0; i <= mu_s.size(); i++)    // **** line 92
        {
          std::vector<double> mu( mu_s.size());
          std::vector<double> phi( mu_s.size());
          std::vector<double> strength( mu_s.size());
          //std::vector<double> eta( mu_s.size());
          std::vector<double> viscous_stress( mu_s.size());

          // Calculate viscous stress
          viscous_stress[i] = 2. * background_viscosities[i] * edot_ii;
```
Note how we run the loop index `i` all the way up to `mu_s.size()` (by comparing with `<=` instead of `<`), setting the size of the various arrays to `mu_s.size()`, and then indexing with `[i]`. This means that we are accessing the element past the end.

I believe that the loop should actually only run as in 
```
      for (unsigned int i = 0; i < mu_s.size(); i++)    // **** line 92
```

@alarshi and @naliboff -- you worked on this. Any insight?

**Comment by bangerth** [comment]
Related to #1728.

**Comment by naliboff** [comment]
Indeed, this is now addressed in #1779. Cheers.
